### PR TITLE
Duplicate connection page UI

### PR DESCRIPTION
**Pull request by emilys314**
Towards https://issues.redhat.com/browse/RHOAIENG-10314

## Description

Adds the UI portion of the create / duplicate connection type page. The jira issue is for the duplicate page, but a lot of the create page is already made since it's dependent. 

image
image
image

Error state:
image

## How Has This Been Tested?

Manually on local. Start the front end and back end. Then go to the `/connectionTypes/create` or `/connectionTypes/duplicate/<name>` routes.

`/duplicate` requires an existing connection type. So you can go to `/create` and Make one first.

## Test Impact

Cypress tests were added for create and duplicate pages. 

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by jenny-s51 — COMMENTED** [review]
Duplicate page looks good and works as expected @emilys314 , tested by creating a data connection and was able to duplicate it successfully. 

Just a note about the empty state table under Fields, would we want to use a default empty state rather than an empty state table? According to the [mocks](https://www.figma.com/design/97akqulXiGIGtvxkdPLjMm/Data-connections-enhancement?node-id=1185-124564&t=Viq6w5jAVMlkbFTV-0) it looks like we don't want to show the table headers if there are no fields.

Otherwise LGTM!

**Review by jenny-s51 — APPROVED** [review]
/lgtm

**Comment by ashley-o0o** [comment]
Tested locally, and ran the Cypress tests in the Cypress UI. Everything looking good on my end! 
/lgtm

**Comment by ashley-o0o** [comment]
Tested locally, and ran the Cypress tests in the Cypress UI. Everything looking good on my end! One note, you'll need to squash commits before approval :)
/lgtm

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx
Comment (jeff-phillips-18) [inline frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx:82]: would be better to tighten it up by removing the top padding on this section
```suggestion
        <PageSection className="pf-v5-u-pt-0 isFilled variant="light">
```

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx
Comment (jeff-phillips-18) [inline frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx:105]: Would be better to remove the top margin
```suggestion
            <FormSection className="pf-v5-u-mt-0" title="Fields">
```

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx
Comment (emilys314) [reply frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx]: True i see that in the mocks as well. updated 

<details>
<summary>gif</summary>

ezgif-4-8f5b0eb098

</details>

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx
Comment (emilys314) [reply frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx]: updated

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx
Comment (emilys314) [reply frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx]: updated (update images are in the pr description)

**Comment by emilys314** [comment]
> Tested locally, and ran the Cypress tests in the Cypress UI. Everything looking good on my end! One note, you'll need to squash commits before approval :) /lgtm

I think this is no longer the case. based on what i was seeing in some slack convos,  the PR should squash it for you now

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypeFieldsTableRow.tsx
Comment (christianvogt) [inline frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypeFieldsTableRow.tsx:37]: Must wrap `Text` with `<TextContent>`

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx
Comment (emilys314) [reply frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx:104]: Okay sounds good, I wasn't sure since <Text> adds some extra css styles too.

So do you mainly use the <Text> if you have a new line / section of text? (like in the `./CreateConnectionTypeFieldsTableRow.tsx` for the field-name column?)

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypeFieldsTableRow.tsx
Comment (emilys314) [reply frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypeFieldsTableRow.tsx:40]: I actually can't replicate this (from api response fields or the mock object) on firefox or chrome.

What's the intended behavior? to make new lines?

Comment (christianvogt) [reply frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypeFieldsTableRow.tsx:40]: I think it has to do with an error on my side in how i serialized the test model. So ignore this.

File: frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx
Comment (christianvogt) [reply frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx:104]: If we just want to inherit the styles from the parent just wrap in a `div` or render plain text.
I don't think we specifically need a proper `p` tag here.

File: frontend/src/pages/connectionTypes/create/CreateConnectionTypePage.tsx
Comment (christianvogt) [inline frontend/src/pages/connectionTypes/create/CreateConnectionTypePage.tsx:66]: You should clear the error before submitting:
```suggestion
      setIsCreateLoading(true);
      setCreateError(null);
      const response = await createConnectionType(previewConnectionTypeObj);
      if (response.error) {
        setCreateError(response.error);
```

File: frontend/src/pages/connectionTypes/create/DuplicateConnectionTypePage.tsx
Comment (christianvogt) [inline frontend/src/pages/connectionTypes/create/DuplicateConnectionTypePage.tsx:20]: `username` should be set to the user name of the logged in user.

**Pull request merged**
